### PR TITLE
feat: add alert parser route

### PR DESCRIPTION
**Pull request by ismay**
- [x] Use existing userGroup component
- [x] Display notification for unexpected errors

### CODE REVIEW

File: src/commands/CommandEditAlertParserForm.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                        <SingleSelectField
+                            selected={userGroup.name}
+                            label={i18n.t('User Group')}
+                            disabled
+                        >
+                            <SingleSelectOption
+                                label={userGroup.name}
+                                value={userGroup.name}
+                            />
+                        </SingleSelectField>
```
Comment (HendrikThePendric) [inline src/commands/CommandEditAlertParserForm.js:132]: is this really a select with one option? Why do we need that?

**Review by HendrikThePendric — APPROVED** [review]
LGTM, left one question

File: src/commands/CommandEditAlertParserForm.js (hunk 1 above)
Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js:132]: Yeah it's the field that the comment up top is referring to. It's a dummy field that just shows the current value. I just took it from the old app, but maybe I'm misinterpreting the purpose. To me it seemed like it was just there to show the current value

Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js:132]: Screenshot 2020-08-26 at 17 22 54

Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js:132]: Screenshot 2020-08-26 at 17 23 26

Comment (Mohammer5) [reply src/commands/CommandEditAlertParserForm.js:132]: I'm doing the same, but I would probably use the existing `FieldUserGroup`, then you have it in your form state.
It seems like not providing the value is still working, but I'd prefer to be explicit here.

Not that important for now though, I'll align all edit forms after the release anyways

Comment (HendrikThePendric) [reply src/commands/CommandEditAlertParserForm.js:132]: 👍  thanks for clearing that up!

File: src/commands/CommandEditAlertParserForm.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    const history = useHistory()
+    const engine = useDataEngine()
+    const updateCommand = command =>
+        engine
+            .mutate(mutation, { variables: { command, id } })
+            .then(() => {
+                history.push(SMS_COMMAND_LIST_PATH)
+            })
+            .catch(error => {
+                const isValidationError = error.type === 'access'
```
Comment (HendrikThePendric) [inline src/commands/CommandEditAlertParserForm.js:60]: I'm not that familiar with the new error codes the webapi is producing, but at first glance this doesn't make sense to me.... If the type is `access` I would assume this error is the result of some authentication issue, and not really to do with validation (i.e. sth being wrong with the data)...

File: src/commands/CommandEditAlertParserForm.js (hunk 1 above)
Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js:132]: > I'm doing the same, but I would probably use the existing FieldUserGroup, then you have it in your form state.
It seems like not providing the value is still working, but I'd prefer to be explicit here.
>
> Not that important for now though, I'll align all edit forms after the release anyways

Ah I missed that that already existed. I'll use that one instead then 👍

File: src/commands/CommandEditAlertParserForm.js (hunk 2 above)
Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js:60]: Yeah I agree. This is what Austin said to use to catch validation type errors. I'd have to look into the data engine source, but it made sense to me at the time for the range of error codes associated with `access`.

File: src/commands/CommandEditAlertParserForm.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        resource: 'smsCommands',
+        id: ({ id }) => id,
+        params: {
+            fields: [
+                'name',
+                'parserType',
+                'receivedMessage',
+                'userGroup[name]',
+            ],
+            paging: 'false',
```
Comment (HendrikThePendric) [inline src/commands/CommandEditAlertParserForm.js:37]: I think there are two things wrong here:
- `'false'` (string) should be `false` (boolean)
- But the bigger problem is that we are fetching a single item here, so the `paging` property seems irrelevant

Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js:37]: > 'false' (string) should be false (boolean)

You're right, I usually use a bool here as well. I blindly copied that from another area in the app. I think we're doing that all over the app. We should change that.

> But the bigger problem is that we are fetching a single item here, so the paging property seems irrelevant

Yeah we should be able to just leave it out here.

Comment (HendrikThePendric) [reply src/commands/CommandEditAlertParserForm.js:37]: 👍  just to confirm, I'm leaving it out and it's working fine

Comment (Mohammer5) [reply src/commands/CommandEditAlertParserForm.js:37]: I tried that, didn't work, I used `false` first before as well...

File: src/commands/CommandEditAlertParserForm.js (hunk 1 above)
Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js]: Ok, should be good now.

File: src/commands/CommandEditAlertParserForm.js (hunk 3 above)
Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js]: Removed the paging.

Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js]: Oh because I'm using the boolean in the scheduler as well. I think it works there. Let me check

Comment (ismay) [reply src/commands/CommandEditAlertParserForm.js]: Yeah here for example: https://github.com/dhis2/scheduler-app/pull/23/files#diff-93586d896c98cc73ff851bfabac6e6eaR19. This seems to work

File: src/commands/FieldCommandConfirmMessage.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import { FIELD_COMMAND_CONFIRM_MESSAGE_NAME } from './fieldNames'
+import { dataTest } from '../dataTest'
+import i18n from '../locales'
+
+const { Field } = ReactFinalForm
+
+export const FieldCommandConfirmMessage = () => (
+    <Field
+        dataTest={dataTest('forms-fieldcommandconfirmmessage')}
```
Comment (Mohammer5) [inline src/commands/FieldCommandConfirmMessage.js:12]: completely irrelevant now (and I assume that you just copied it from where I did it wrong as well):
the data test prop should be `commands-fieldcommandconfirmmessage`

**Review by Mohammer5 — APPROVED** [review]

File: src/commands/FieldCommandConfirmMessage.js (hunk 4 above)
Comment (ismay) [reply src/commands/FieldCommandConfirmMessage.js]: Cool, updated it 👍
